crates/python/src/execution_data.rs

```diff
@@ -37,25 +78,77 @@ py_wrap_type! {
     PyReadoutMap(ReadoutMap) as "ReadoutMap";
 }
 
+// py_wrap_union_enum! {
```

Comment:
Note to self, need to remove this or figure out how to get the macro to support this. `PyArray2` is not clone, hence the manual implementation below.

Reply:
I don't think `PyArray2` should need to be `Clone`. It's the wrapped types, like `RegisterMatrix` that need to be `Clone`.

Most likely, you should be using `Py<PyArray2<_>>`?

Reply:
Yeah, the clone thing was a red herring. Revisited it, and the issue I haven't figured out how to work around is that `PyArray2` doesn't implement `PyTryFrom` for `Array2` and since neither of which are defined in this crate, I can't implement it myself.

Reply:
Might be worth investigating blanket implementations in `rigetti-pyo3` based on `pyo3::{FromPyObject, IntoPy}`.